### PR TITLE
Add release 2 changelog

### PR DESCRIPTION
**Pull request by pbiggar**

### CODE REVIEW

File: docs/reference/changelog.md [hunk 1]
```diff
@@ -4,122 +4,123 @@ title: Changelog
...
+[2](https://blog.darklang.com/why-dark-didnt-choose-rust/),
+[3](https://blog.darklang.com/leaving-ocaml/).
 
-As we prepare to release the new version, we want to give advanced warning about
-some changes to the language and standard library. Though Dark was designed to
-avoid changes to existing functionality, allowing for new versions of the
-language and standard library to be opted-in, we have not built that part of the
-procuct, and given the relatively low usage of Dark we felt the tradeoff was not
-worth it yet.
+As of now, the rewrite has been deployed to much, but not all of our
```
Comment (StachuDotNet) [inline docs/reference/changelog.md:17]:
```suggestion
As of now, the rewrite has been deployed too much, but not all of our
```

File: docs/reference/changelog.md [hunk 2]
```diff
@@ -179,45 +177,70 @@ List.uniqueBy_v0 [1;2;3;4] (fun x -> Int.divide_v0 x 2) = [ 1, 3, 4 ] // old Dar
...
-Runtime errors (including type errors) are not accessible via Dark programs, and
-will always terminate the Dark program when they are accessed. As a result, the
-text of any runtime error should not be accessible to your programs or your
-users.
+- bare field names, such as in `{ id : 5 }`. You need to quote field names:
+  `{ "id": 5 }`
+- newlines and invalid bytes in JSON strings
+- the bare identifiers `NaN`, `Infinity` and `-Infinity` are no longer parsed
+  into valid floats (note that the old version of Dark might have generated JSON
+  with these values in it)
```
Comment (StachuDotNet) [inline docs/reference/changelog.md:225]: Can/should we avail a fn to users which turns such invalid JSON into valid JSON?

File: docs/reference/changelog.md [hunk 3]
```diff
@@ -245,31 +275,78 @@ Minor differences:
   - The `Date` header is now always present
   - The `Server` header is now `darklang` and always present
 
-## Improvements in the 2021 rewrite
+## Testing changes
 
-These are being updated below as we remember them, and will be rewritten into a
-changelog when they ship.
+We have gone from abotu 250 backend tests to over 11,000. We now have custom
```
Comment (StachuDotNet) [inline docs/reference/changelog.md:280]:
```suggestion
We have gone from about 250 backend tests to over 11,000. We now have custom
```

File: docs/reference/changelog.md (hunk 2 above)
Comment (pbiggar) [reply docs/reference/changelog.md:225]: At this point everyone has been switched over, so I don't think anyone would have use for one.

**Pull request merged**